enzyme/Enzyme/TypeAnalysis/TypeAnalysis.cpp

```diff
@@ -3450,19 +3450,48 @@ void TypeAnalyzer::visitMemTransferCommon(llvm::CallBase &MTI) {
     EmitFailure("IllegalUpdateAnalysis", MTI.getDebugLoc(), &MTI, ss.str());
     report_fatal_error("Performed illegal updateAnalysis");
   }
-  res.insert({}, BaseType::Pointer);
-  res = res.Only(-1, &MTI);
-  updateAnalysis(MTI.getArgOperand(0), res, &MTI);
-  updateAnalysis(MTI.getArgOperand(1), res, &MTI);
-#if LLVM_VERSION_MAJOR >= 14
-  for (unsigned i = 2; i < MTI.arg_size(); ++i)
-#else
-  for (unsigned i = 2; i < MTI.getNumArgOperands(); ++i)
-#endif
-  {
-    updateAnalysis(MTI.getArgOperand(i),
-                   TypeTree(BaseType::Integer).Only(-1, &MTI), &MTI);
+  for (unsigned i = 0; i < 2; ++i) {
+    // if (MTI.getArgOperand(i) != origArg)
+    //   continue;
+    if (!MTI.getAttributes().hasParamAttr(i, "enzyme_type")) {
+      updateAnalysis(MTI.getArgOperand(i),
```

Comment:
The changes to TA should be reverrted and this should already be handled here: https://github.com/EnzymeAD/Enzyme/blob/9d6a86b46086a2009725f8bfb9af89e7dc7168f6/enzyme/Enzyme/TypeAnalysis/TypeAnalysis.cpp#L1242

The problem in the test case won't be fixed by TA [since it was written to mean TA wasn't enough for -- but the change needs to happen in visitMemTransfer to also parse the arg TT's and or it in